server/src/test/java/org/elasticsearch/cluster/metadata/DataLifecycleTests.java

```diff
@@ -39,19 +40,16 @@ protected Writeable.Reader<DataLifecycle> instanceReader() {
 
     @Override
     protected DataLifecycle createTestInstance() {
-        if (randomBoolean()) {
-            return new DataLifecycle();
-        } else {
-            return new DataLifecycle(randomMillisUpToYear9999());
-        }
+        return switch (randomInt(2)) {
+            case 0 -> new DataLifecycle();
+            case 1 -> new DataLifecycle.Builder().dataRetention(new DataLifecycle.Retention(null)).build();
+            default -> new DataLifecycle(randomMillisUpToYear9999());
+        };
     }
 
     @Override
     protected DataLifecycle mutateInstance(DataLifecycle instance) throws IOException {
-        if (instance.getDataRetention() == null) {
-            return new DataLifecycle(randomMillisUpToYear9999());
-        }
-        return new DataLifecycle(instance.getDataRetention().millis() + randomMillisUpToYear9999());
+        return randomValueOtherThan(instance, this::createTestInstance);
```

Comment:
I've seen this used somewhere else, too. But doesn't it violate the spirit of this method in two ways?

1. The javadoc says `Returns an instance which is mutated slightly`. In this case it's completely random, so not just a slight change
2. (The bigger one) The test guaranteed to work even if equals is broken (in certain ways), isn't it? This method is saying "give me an object that you want to test is not equal to the given one". And we're saying "here is an object that I guarantee is not equal to the given object". Say equals is broken and returns `true` in a case where the objects are not supposed to be equal. This method would never allow that (the one thing you really want to be testing) to be returned because `randomValueOtherThan` would think that it's equal.

Reply:
You are right, I was a bit lazy in this one, I will write a better one. 

About the second point, equality is not the only thing we are testing in this situations, for example serialisation could also be broken here. 

But in any case, you are right and I will fix it :)